exec/java-exec/src/test/java/org/apache/drill/exec/client/DrillClientSystemTest.java

```diff
@@ -73,4 +77,90 @@ public void testSubmitPlanTwoNodes() throws Exception {
+    endpoint = endpointsList.iterator().next();
+    assert(endpointsList.size() == 1);
+    assert(endpoint.getAddress().equalsIgnoreCase(drillBitConnection));
+    assert(endpoint.getUserPort() == client.getConfig().getInt(ExecConstants.INITIAL_USER_PORT));
+
+    // Test with single drillbit ip:port
+    endpointsList.clear();
+    drillBitConnection = "10.10.100.161:5000";
+    String[] ipAndPort = drillBitConnection.split(":");
+    client.populateEndpointsList(endpointsList, drillBitConnection);
+    assert(endpointsList.size() == 1);
+
+    endpoint = endpointsList.iterator().next();
+    assert(endpoint.getAddress().equalsIgnoreCase(ipAndPort[0]));
+    assert(endpoint.getUserPort() == Integer.parseInt(ipAndPort[1]));
+
+    // Test with multiple drillbit ip
+    endpointsList.clear();
+    drillBitConnection = "10.10.100.161,10.10.100.162";
+    client.populateEndpointsList(endpointsList, drillBitConnection);
+    assert(endpointsList.size() == 2);
+
+    endpointIterator = endpointsList.iterator();
+    endpoint = endpointIterator.next();
+    assert(endpoint.getAddress().equalsIgnoreCase("10.10.100.161"));
+    assert(endpoint.getUserPort() == client.getConfig().getInt(ExecConstants.INITIAL_USER_PORT));
+
+    endpoint = endpointIterator.next();
+    assert(endpoint.getAddress().equalsIgnoreCase("10.10.100.162"));
+    assert(endpoint.getUserPort() == client.getConfig().getInt(ExecConstants.INITIAL_USER_PORT));
+
+    // Test with multiple drillbit ip:port
+    endpointsList.clear();
+    drillBitConnection = "10.10.100.161:5000,10.10.100.162:5000";
+    client.populateEndpointsList(endpointsList, drillBitConnection);
+    assert(endpointsList.size() == 2);
+
+    endpointIterator = endpointsList.iterator();
+    endpoint = endpointIterator.next();
+    assert(endpoint.getAddress().equalsIgnoreCase("10.10.100.161"));
+    assert(endpoint.getUserPort() == 5000);
+
+    endpoint = endpointIterator.next();
+    assert(endpoint.getAddress().equalsIgnoreCase("10.10.100.162"));
+    assert(endpoint.getUserPort() == 5000);
+
+    // Test with multiple drillbit with mix of ip:port and ip
+    endpointsList.clear();
+    drillBitConnection = "10.10.100.161:5000,10.10.100.162";
+    client.populateEndpointsList(endpointsList, drillBitConnection);
+    assert(endpointsList.size() == 2);
+
+    endpointIterator = endpointsList.iterator();
+    endpoint = endpointIterator.next();
+    assert(endpoint.getAddress().equalsIgnoreCase("10.10.100.161"));
+    assert(endpoint.getUserPort() == 5000);
+
+    endpoint = endpointIterator.next();
+    assert(endpoint.getAddress().equalsIgnoreCase("10.10.100.162"));
+    assert(endpoint.getUserPort() == client.getConfig().getInt(ExecConstants.INITIAL_USER_PORT));
+
+    // Test with empty string
+    endpointsList.clear();
+    drillBitConnection = "";
+    client.populateEndpointsList(endpointsList, drillBitConnection);
+    assert(endpointsList.size() == 0);
+
+
```

Comment:
Test the other strange cases noted in earlier comments: spaces, non-numeric ports, missing host, etc:

    " foo : 10 ", "foo: ", "foo:", "foo:bar", ":20", etc.

for errors:

    {
        try { parseEndpoints( "foo:bogus" ); fail( ); }
        catch( YourException e ) { }
    }